### PR TITLE
Add "Deploy & the Deploy Lifecycle" concept

### PR DESCRIPTION
**Pull request by andrzej-casper**
### What does this PR introduce?

New page "Deploy & the Deploy Lifecycle" under _Concepts_.

### Additional context

These changes are backported from the default branch of [docs-new](https://github.com/casper-network/docs-new) repository, so you can preview them [live here](https://docs-new.casper.network/deploy-and-deploy-lifecycle/).

Original PR authored by @caspermel: https://github.com/casper-network/docs-new/pull/145.

### Checklist

- [x] I ran the docs locally using `yarn install` and `yarn run start`.
- [x] All links (internal and external) have been verified.
- [x] All technical procedures have been tested.

### CODE REVIEW

**Review by ACStoneCL — CHANGES_REQUESTED** [review]
I'm fine with extracting this from the Casper Network overview, but can we use the same PR to remove the duplicate content from [that page?](https://docs.casperlabs.io/concepts/design/casper-design/#execution-semantics-deploys)

File: source/docs/casper/concepts/deploy-and-deploy-lifecycle.md [hunk 1]
```diff
@@ -1,7 +1,79 @@
+---
+title: Deploy & Deploy Lifecycle
```
Comment (ipopescu) [inline source/docs/casper/concepts/deploy-and-deploy-lifecycle.md:2]:
```suggestion
title: Deploys and the Deploy Lifecycle
```

File: source/docs/casper/concepts/deploy-and-deploy-lifecycle.md [hunk 2]
```diff
@@ -1,7 +1,79 @@
...
+slug: /deploy-and-deploy-lifecycle
+---
+
 # Deploy & the Deploy Lifecycle
```
Comment (ipopescu) [inline source/docs/casper/concepts/deploy-and-deploy-lifecycle.md:6]:
```suggestion
# Deploys and the Deploy Lifecycle
```

File: source/docs/casper/concepts/deploy-and-deploy-lifecycle.md [hunk 3]
```diff
@@ -1,7 +1,79 @@
...
+- Body: Containing payment code and session code (more details on these below)
+- Header: containing
+    - The [Public Key](./serialization-standard.md#publickey) of the account in whose context the deploy will run
+    - The timestamp of the deploy’s creation
+    - A time-to-live, after which the deploy expires and cannot be included in a block
+    - the `blake2b256` hash of the body
+- Deploy hash: the `blake2b` hash of the Header
+- Approvals: the set of signatures which have signed the deploy hash; these are used in the [account permissions model](./design/casper-design.md#accounts-associated-keys-weights)
+
+## Deploy Lifecycle {#execution-semantics-phases}
```
Comment (ipopescu) [inline source/docs/casper/concepts/deploy-and-deploy-lifecycle.md:25]:
```suggestion
## The Deploy Lifecycle {#execution-semantics-phases}
```

**Review by ipopescu — COMMENTED** [review]
I left some small nits in the headings. As @ACStoneCL requested, please remove the content from the original location. Otherwise, it looks good!

**Comment by caspermel** [comment]
Yes, I agree that the original content should be removed.

**Pull request merged**